ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -103,16 +105,16 @@ class Endpoints(
     }
     import scalaz.std.partialFunction._, scalaz.syntax.arrow._
     (dispatch &&& { case r => r }) andThen { case (lcFhr, req) =>
-      extendWithRequestIdLogCtx(implicit lc =>
-        // TODO: Refactor this somehow into an own function
-        for {
-          _ <- Future.unit
-          _ = logger.trace(s"Incoming request on ${req.uri}")
-          t0 = System.nanoTime()
-          res <- lcFhr(lc)
-          _ = logger.trace(s"Processed request after ${System.nanoTime() - t0}ns")
-        } yield res
-      )
+      extendWithRequestIdLogCtx(implicit lc => {
+        val t0 = System.nanoTime
+        logger.trace(s"Incoming request on ${req.uri}")
+        Timed
```

Comment:
I think this is perfectly fine to setup the infrastructure in this PR but in terms of general metrics, I think we need to be more specific. There is no meaningful correlation between times on queries, package downloads or command submissions which is usually an indicator that we need more specific metrics. I think we can address all of that in a separate PR provided we hide the option to avoid having backwards compat issues around this one.

Reply:
As proposed, will do this in a seperate PR.